docs/copilot/guides/mcp-developer-guide.md

```diff
@@ -0,0 +1,259 @@
+
+![Screenshot that shows the MCP Resources Quick Pick.](images/mcp-developer-guide/mcp-resources-picker.png)
+
+VS Code supports resource updates, enabling users to see changes to the contents of a resource in real-time in the editor.
+
+#### Resource templates
+
+VS Code also supports [resource templates](https://modelcontextprotocol.io/docs/concepts/resources#resource-templates), enabling users to provide input parameters when referencing a resource. For example, a database query tool could ask for the database table name.
+
+When accessing a resource with a template, users are prompted for the required parameters in a Quick Pick. You can provide completions to suggest values for the parameter.
+
+### Prompts
+
+Prompts are reusable chat prompt templates that users can invoke in chat by using a slash command (`mcp.servername.promptname`). Prompts can be useful for onboarding users to your servers by highlighting various tools or providing built-in complex workflows that adapt to the user's local context and service.
+
+If you define [completions](https://modelcontextprotocol.io/specification/2025-06-18/server/utilities/completion) to suggest values for prompt input arguments, then VS Code shows a dialog to collect input from the user.
+
+```typescript
+server.prompt(
+    "teamGreeting", "Generate a greeting for team members",
+    {
+        name: completable(z.string(), (value) => {
+            return ["Alice", "Bob", "Charlie"].filter(n => n.startsWith(value));
+        })
+    },
+    async ({ name }) => ({
+        messages: [{
+            role: "assistant",
+            content: { type: "text", text: `Hello ${name}, welcome to the team!` }
+        }]
+    })
+);
+```
+
+![Screenshot that shows the prompt dialog for an MCP prompt with input parameters.](images/mcp-developer-guide/mcp-prompt-argument.png)
+
+> [!NOTE]
+> Users can enter a terminal command in the prompt dialog and use the command output as input for the prompt.
+
+When you include a resource type in the prompt response, VS Code attaches that resource as context to the chat prompt.
+
+### Authorization
+
+VS Code supports MCP servers that require authentication, allowing users to interact with an MCP server that operates on behalf of their user account for that service.
+
+The [authorization specification](https://modelcontextprotocol.io/specification/2025-06-18/basic/authorization) cleanly separates MCP servers as Resource Providers from Authorization Servers, allowing developers to delegate authentication to existing identity providers rather than building their own OAuth implementations from scratch.
+
+VS Code supports the following client functionality:
+
+- The MCP server behaves as an authorization server
+- The MCP server behaves as a resource server
```

Comment:
The top one is an "old spec" concept and really a deprecated idea. Can we deprioritize that one? A lot of existing MCP servers do use it still so we should still call it out